LEDControl_v1/MCAL/gpio/gpio_program.c

```diff
@@ -0,0 +1,379 @@
+ *		@author			Ahmad Hesham
+ *      @email			ahmedhesham606@icloud.com
+ */
+
+/**----------------------------------------------------------
+ *  GLOBAL FUNCTION IMPLEMENTATIONS
+ *----------------------------------------------------------*/
+#include "gpio_interface.h"
+#include "gpio_private.h"
+#include "TM4C123.h"
+
+ptr_gpio_callback_t_ gpiof_callback = NULL_PTR;
+
+/**----------------------------------------------------------
+ *  GLOBAL FUNCTION IMPLEMENTATIONS
+ *----------------------------------------------------------*/
+ /**
+  *	@brief			This function is used to initialize a specific pin
+  *	@param [in]		ptr_str_gpio_config				:	Ptr to the gpio instance structure
+  *	@param [out]	none
+  *	@return			ERROR_OK						:	In case of successeion
+  *	@return			GPIO_INVALID_PIN_INDEX			:	In case of wrong pin index
+  *	@return			GPIO_INVALID_PIN_MODE			:	In case of wrong mode choosen
+  *	@return			GPIO_INVALID_PIN_DIRECTION		:	In case of wrong direction choosen
+  *	@return			GPIO_INVALID_OP_CURRENT			:	In case of wrong current choosen
+  *	@return			GPIO_INVALID_INTERNAL_ATTACH	:	In case of wrong internal attach
+  */
+enu_error_status_t_ gpio_pin_init(str_gpio_config_t_* ptr_str_gpio_config)
+{
+	enu_error_status_t_ enu_ret_val = ERROR_OK;
+
+	uint8_t_ uint8_pin_num = 0, uint8_port_num = 0;
+	uint8_pin_num = ptr_str_gpio_config->enu_pin_index % 10;
+	uint8_port_num = ptr_str_gpio_config->enu_pin_index / 10;
+
+	if (uint8_port_num > 5 || uint8_pin_num > 7)
+	{
+		enu_ret_val = GPIO_INVALID_PIN_INDEX;
+	}
+	else
+	{
+		if (((5 == uint8_port_num) && (uint8_pin_num > 4)) || ((4 == uint8_port_num) && (uint8_pin_num > 5)))
+		{
+			enu_ret_val = GPIO_INVALID_PIN_INDEX;
+		}
+		else
+		{
+			/*STEP 1: Enable clock to GPIO Port Used using RCGCGPIO register*/
+			if (1 == GET_BIT(RCGCGPIO_, uint8_port_num))
+			{
+				/*ENABLED*/
+			}
+			else
+			{
+				SET_BIT(RCGCGPIO_, uint8_port_num);
+			}
+
+			/*STEP 2: Assigning pin direction, Output Current, and Pin Internal Attach*/
+			if (GPIO_OUTPUT == ptr_str_gpio_config->enu_pin_direction)
+			{
+				SET_BIT(GPIODIR(uint8_port_num), uint8_pin_num);
+				
+				/*STEP 3: Assigning whether the pin is working digital or Alternative function*/
+				switch (ptr_str_gpio_config->enu_pin_mode)
+				{
+				case GPIO_DIO:
+					GPIOLOCK(uint8_port_num) = GPIO_UNLOCKING_VALUE;	GPIOCR(uint8_port_num) = 0x1;
+					CLR_BIT(GPIOAFSEL(uint8_port_num), uint8_pin_num);
+					
+					break;
+				default:
+					GPIOLOCK(uint8_port_num) = GPIO_UNLOCKING_VALUE;	GPIOCR(uint8_port_num) = 0x1;
+					SET_BIT(GPIOAFSEL(uint8_port_num), uint8_pin_num);
+					GPIOLOCK(uint8_port_num) = GPIO_UNLOCKING_VALUE;	GPIOCR(uint8_port_num) = 0x1;
+					(GPIOPCTL(uint8_port_num) &= (~((uint32_t_)15 << (4 * uint8_pin_num))));
+					(GPIOPCTL(uint8_port_num) |= ((uint32_t_)ptr_str_gpio_config->enu_pin_mode << (4 * uint8_pin_num))); 	
+					break;
+				}
+				
+
+				switch (ptr_str_gpio_config->enu_pin_output_current)

```

Comment:
switch case shouldn't have more than one break

Reply:
Show me a reference for this role.